### ISSUE TITLE
Chromium-based browsers since v81 may cause unexpected DOM manipulation effects

### ISSUE THREAD

**Issue by lyubomyr-shaydariv** [jquery/jquery#4679]
### Description ###

Chromium / Chrome 81.0.4044.113 and later (at least up to Chromium 84 dev I could test on) seem to break the way jQuery maniplates DOM in certain cases. I can reproduce this behavior in a [multiselect](https://github.com/crlcu/multiselect) jQuery plugin that reorders DOM nodes if the dedicated sorting option is set only.

####  What do you expect to happen?

Re-appended nodes are appended in the expected order, also prior to Chrome 81.

####  What actually happens?

It depends: some DOM nodes (perhaps `option` only?) are not rendered (unless the parent `select` is clicked or selects all items), or get in unexpected order.

#### Which browsers are affected?

Chromium-based browsers since 81. Other browsers, i.e. Firefox, work fine.

### Link to test case ###

Can't provide a runnable test-case on JSFiddle or JSBin because GitHub raw content cannot be hot-linked and serve as CDN. Please see the case in the original issue https://github.com/crlcu/multiselect/issues/194 . The test case attempts to reorder child `option` elements that relies on the plugin sorting function, basically simple

```js
this.children()
    .sort(callback)
    .appendTo(this);
```

The `appendTo` actually prevents the plugin from working fine. Rewriting/patching the plugin `$.fn.mSort` method in the following manner:

```js
var children = this
    .children()
    .sort(callback);
for ( var j = 0; j < this.length; j++ ) {
    for ( var i = 0; i < children.length; i++ ) {
        this[j].appendChild(children[i]);
    }
}
```

fixes the issue. If the plain"-JS-DOM-append" code is rewritten with jQuery (say, `this.append(child)` for each child), it won't fix, so `appendTo` is not the only method with the defect.

Related issues (not sure if they are caused with DOM nodes reordering): https://github.com/crlcu/multiselect/issues/192 and https://github.com/crlcu/multiselect/issues/193 .

What might cause these methods to have unexpected behavior in Chrome 81 and later?

----

##### Edit 1

Removing the multiselect plugin narrows down the issue test case: https://jsfiddle.net/2yzneg8d/

----

##### Edit 2

Removing the `sort` method call removes all children from the select box.

**Comment by mgol** [comment]
Thanks for the report! This looks like a Chrome bug, already reported in https://bugs.chromium.org/p/chromium/issues/detail?id=1072475. Let's keep this jQuery issue open to track it. I suspect we'll just want to rely on an upstream fix here.

**Comment by mgol** [comment]
The issue is reproducible in Chrome 81 without any external library: https://codepen.io/mgol/pen/eYpBqMy

**Comment by mgol** [comment]
BTW, this fails even with jQuery 1.x/2.x, this really needs to be fixed on the Chromium side.

**Comment by jmontoyaa** [comment]
I confirm that code fixes the issue in Chrome v83 (beta)

**Comment by mgol** [comment]
This has been fixed in Chrome 84 (currently in Canary). We'll see whether they backport it to v83 & v81.

**Comment by mgol** [comment]
It looks like they backported it to versions 83 & 81. We just need to wait for the next 81 update now.

**Comment by mgol** [comment]
The fix was released to stable Chrome in 81.0.4044.138, other Chromium-based browsers should follow shortly. Closing as solved.
